### PR TITLE
Fix tensorboard logging, and add tboard logging consumable by the OSS metrics handler

### PR DESCRIPTION
**Pull request by taylanbil**
* added a new tensorboard log wrapper progress bar which behaves in a
compatible way w/ the metrics handler
  * has one summary writer, called "run" (great naming choice by me)
  * prefixes the stats collected by fairseq trainers (such as loss, wall
time, gradient norms etc) by the "tag" passed to the progress_bar
    * e.g.: metric name is "loss", "tag" is "train", it logs
  * logs xla metrics at the end of epochs.
"train-loss" to tensorboard. similarly, "validation-loss", etc.

* if `--metrics_debug` and `--tensorboard-logdir` are passed in, then
`train.py` will use the said progress_bar above.

* call `item()` before going to tensorboardX, as suggested by tensorboardX

### CODE REVIEW

**Comment by taylanbil** [comment]
Example output [here](https://screenshot.googleplex.com/HyCzmAzBNN7)

**Review by dlibenzi — APPROVED** [review]

**Review by zcain117 — COMMENTED** [review]
Does this version of tensorboard support writing to GCS?

**Review by zcain117 — APPROVED** [review]

**Comment by taylanbil** [comment]
@zcain117 This code is using tensorboardX, iirc we verified that it does support gcs. Isn't that what we use for imagenet tests?

**Comment by zcain117** [comment]
> @zcain117 This code is using tensorboardX, iirc we verified that it does support gcs. Isn't that what we use for imagenet tests?

Yup, that's the right one. Thanks for checking
